### PR TITLE
Add a note about org loading limitation

### LINKED ISSUE #3277: Add FAQ about Org loading limitations
Since Spacemacs uses the org version from the org ELPA repository we must inform users that they should not define Org related stuff that require Org to be loaded before `dotspacemacs/user-config`.

### PR DESCRIPTION
**Pull request by StreakyCobra**
Add a release note about the limitation of org loading since 0.105, and
also document it in the org-layer's README.

Close #3277

### CODE REVIEW

File: core/info/release-notes/0.105.txt [hunk 1]
```diff
@@ -1 +1,6 @@
-Nothing for now.
+1. Org
+
+Spacemacs now uses the `org` version from the org ELPA repository instead of the
+one shipped with emacs. Then, any `org` related code should not be loaded before
+`dotspacemacs/user-config`, otherwise both versions will be loaded and will
+conflict.
```
Comment (syl20bnr) [inline core/info/release-notes/0.105.txt]: Actually this change has been made in 0.104 ;-) I guess we should document it in the FAQ instead.

Comment (robbyoconnor) [inline core/info/release-notes/0.105.txt]: @StreakyCobra -- just move this to `doc/FAQ.org` -- I agree with @syl20bnr

**Comment by StreakyCobra** [comment]
Documented in the FAQ :)

**Comment by robbyoconnor** [comment]
:100: :+1: No issues from me.

**Comment by syl20bnr** [comment]
Thank you again :+1: 
Cherry-picked into `develop` branch, you can safely delete your branch.

**Pull request closed without merging**